sw/device/tests/sram_ctrl_scrambled_access_test.c

```diff
@@ -0,0 +1,205 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/*
+ * For the purpose of this test, we need to read and write to specific
+ * RAM addresses. The semantics are identical to MMIO accesses, however, to
+ * avoid confusion we define a set of "data_region" aliases that do exactly
+ * what MMIO counterparts do, but should be less confusing to a reader
+ * of this code.
+ */
+typedef mmio_region_t data_region_t;
+#define data_region_from_addr mmio_region_from_addr
+#define data_region_read32 mmio_region_read32
+#define data_region_write32 mmio_region_write32
```

Comment:
I am not entirely sure this is necessary. If you think it is, we need to formalize it in a way that these defines are used to access all memories in the chip (flash, otp, etc) in our tests, otherwise we will end up with mixed use of `mmio_region_*` and `data_region_*` which might confuse the readers more. My preference is to keep it simple and stick to `mmio_*` for everything.

Reply:
Done.